span-processing-utils/src/main/java/org/hypertrace/config/span/processing/utils/SpanFilterMatcher.java

```diff
@@ -59,6 +72,12 @@ public boolean matchesServiceName(SpanFilter spanFilter, String serviceName) {
     }
   }
 
+  private boolean hasNoRelationalFilters(LogicalSpanFilterExpression logicalSpanFilterExpression) {
+    return logicalSpanFilterExpression.getOperandsCount() == 0
```

Comment:
This should recurse. Also the operand count is redundant, since if it was 0 none match would return true.

Reply:
Will fix in the later PR. This is known problem. Environment and servicename is not checked at more than 1 nested levels